client.go

```diff
@@ -0,0 +1,153 @@
+package support
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	neturl "net/url"
+)
+
+type HijackHttpOptions struct {
+	Method             string
+	Url                string
+	Success            chan struct{}
+	DockerTermProtocol bool
+	InputStream        io.Reader
+	ErrorStream        io.Writer
+	OutputStream       io.Writer
+	Data               interface{}
+	Header             http.Header
+	Log                Logger
+}
+
+// HijackHttpRequest performs an HTTP  request with given method, url and data and hijacks the request (after a successful connection) to stream
+// data from/to the given input, output and error streams.
+func HijackHttpRequest(options HijackHttpOptions) error {
+	if options.Log == nil {
```

Comment:
So some options are optional? Should we state so in the struct or do more checks here?